ls1intum/Artemis
# [Feature] Add rating to programming exercises

### Checklist
- [x] I tested *all* changes and *all* related features locally and on test server1

### Motivation and Context


As the ratings were missing for programming exercises, we add them in this PR.

### Description

- Display RatingsComponent under the ComplaintInteractionComponent (as in the other exercise types)
- Add correct router link for programming exercises

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise with manual assessment, participate and assess  or use already assessed exercise
4. Navigate to the exercise details view and scroll down (see screenshot below)
5. Add your rating
6. Login as Instructor
7. Go to `Instructor Course Dashboard` and click on Ratings
8. The exercise you rated before should be now displayed here.
9. Click on `Show the submission` and you should be forward to the code editor (make sure it is the correct submission)

<img width="817" alt="image" src="https://user-images.githubusercontent.com/41108487/100680688-68a71200-3372-11eb-8fd3-5229e31319e7.png">

### Test Coverage



Nothing changed

### Screenshots



<img width="801" alt="image" src="https://user-images.githubusercontent.com/41108487/100679740-6cd23000-3370-11eb-872b-79c9cb9bd53b.png">

<img width="1647" alt="image" src="https://user-images.githubusercontent.com/41108487/100679782-84111d80-3370-11eb-88a1-946663260014.png">
